geode-core/src/main/java/org/apache/geode/distributed/internal/DistributionImpl.java

```diff
@@ -181,9 +181,9 @@ public void start() {
       throw new GemFireSecurityException(e.getMessage(),
           e);
     } catch (MembershipConfigurationException e) {
-      throw new GemFireConfigException(e.getMessage());
+      throw new GemFireConfigException(e.getMessage(), e.getCause());
```

Comment:
You should just use the exception here instead of `e.getCause()` otherwise you're discarding one of the stack frames in the exception chain.

Reply:
Wouldn't the message then be duplicated? I am asking out of ignorance :S
I mean probably it should be something like:
```java
throw new GemFireConfigException("Configuration exception while starting up distribution", e);
```
?

Reply:
The actual _message_ (string) would be duplicated, but that's OK. You wouldn't be losing part of the full stack trace however. It's not a big deal since you're typically only concerned with the root cause, but it's always better to leave information which may be helpful for debugging.
And, yes your example `throw` is what I'd suggest. That's also what happens just a few lines higher up.